### DISCUSSION TITLE
Fundamentos da Orientação a Objetos / Mão na massa / Interagindo com listas

### DISCUSSION THREAD

**Discussion by rogerfsp** [balta-io/forum#405] [Tira Dúvidas] [answered]
Prezados,
Agradeço o envio do material sobre Nullable Types, mas continuo cheio de dúvidas sobre este assunto.
Como na situação a seguir, na aula citada no título deste post:

**Considerando a classe:**

_namespace Balta.ContentContext
{
    public class CareerItem
    {
        public CareerItem(int porder, string ptitle, string pdescription, Course pinstanciaCourse)
        {
            Order = porder;
            Title = ptitle;
            Description = pdescription;
            InstanciaCourse = pinstanciaCourse;
        }
        public int Order { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public Course InstanciaCourse { get; set; }
    }
}_

Por que o null fica sublinhado de amarelo (Cannot convert null literal to non-nullable reference type.) na seguinte linha no Program.cs?  :

      var vCarrerItem = new CareerItem(1, "Comece por aqui", "", null);

Me perdoem se minhas dúvidas parecem óbvias, mas como já comentei, sou burro.net e para mim fica difícil entender. Ainda mais quando não ocorre nos vídeos do curso.

Muito obrigado.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @rogerfsp, como vai?

No caso, você determinou que o parâmetro `Course pinstanciaCourse` não seria um nullable e posteriormente passou um valor nulo para ele:
```csharp
var vCarrerItem = new CareerItem(1, "Comece por aqui", "", null);
```

Neste caso, ou você muda de `Course pinstanciaCourse` para `Course? pinstanciaCourse`, ou deixa de enviar `null` como argumento.
No curso de Fundamentos do C# também explico isto, tem uma aula sobre Nullable Types lá.

**Reply by rogerfsp** [reply]
Olá.
Adicionei o ? em Course. Feito isto, a class Program "desamarelou"  e  "amarelou" a classe CareerItem.
Obs. 1: Se eu não envio o null como argumento, ocorre erro, porque CareerItem pede 4 argumentos.
Obs. 2: Na aula "Fundamentos da Orientação a Objetos / Interagindo com listas", aos 3:19,   você mesmo informa null como argumento, mas seu código não fica amarelo  (configuração?).

Favor analisar. Muito obrigado.

**Reply by andrebaltieri** [reply]
Bom dia @rogerfsp, por favor, veja se este vídeo te ajuda:
https://www.youtube.com/watch?v=f4TfIJMRHJo

**Comment by rogerfsp** [comment]
Olá André,

Comentei lá no GitHub. Observe o seguinte:
[image: image.png]

Obrigado.

Em dom., 8 de out. de 2023 às 11:20, Andre Baltieri <
***@***.***> escreveu:
